recipes/theora/all/conanfile.py

```diff
@@ -1,182 +1,96 @@
-from conan.tools.microsoft import msvc_runtime_flag
-from conans import ConanFile, MSBuild, AutoToolsBuildEnvironment, tools
-import functools
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps, cmake_layout
+from conan.tools.files import copy, get
+from conan.tools.microsoft import is_msvc
+from conan.errors import ConanException
 import os
-import re
-import shutil
-import stat
-
-required_conan_version = ">=1.36.0"
 
+required_conan_version = ">=1.52.0"
 
 class TheoraConan(ConanFile):
     name = "theora"
-    description = "Theora is a free and open video compression format from the Xiph.org Foundation"
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/xiph/theora"
-    topics = ("theora", "video", "video-compressor", "video-format")
-    license = "BSD-3-Clause"
-
-    settings = "os", "arch", "compiler", "build_type"
-    options = {
-        "shared": [True, False],
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "shared": False,
-        "fPIC": True,
-    }
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    description = "Theora is a free and open video compression format from the Xiph.org Foundation"
+    topics = "video", "video-compressor", "video-format"
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
 
-    @property
-    def _user_info_build(self):
-        return getattr(self, "user_info_build", self.deps_user_info)
+    # original theora.def from: "https://raw.githubusercontent.com/xiph/theora/v1.1.1/lib/theora.def"
+    # edited to change library name to just "theora"
```

Comment:
we should not be changing things like that 🤔

Reply:
I explained my reasoning here: https://github.com/conan-io/conan-center-index/pull/13452#discussion_r996415036
happy to change with guidance ...